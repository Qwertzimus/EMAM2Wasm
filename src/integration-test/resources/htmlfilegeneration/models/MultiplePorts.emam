package models;

component MultiplePorts {
  ports
    in Q(0 m : 0.1 m : 10 km)^{2,3} m1,
    in Q(5 Hz : 20 kHz)^{3,1} m2,
    in Q(-2 mm/h : 2 km/s) c,
    out Q(-oo km/h : oo km/h)^{2,1} v,
    out Q(-oo km^-1 : oo m^-1) k;

  implementation Math {
    v = (m1*m2) + c;
    Q temp = 0;
    for i=1:3
      temp = temp + m2(i,1)*c;
    end
    k = temp;
  }
}