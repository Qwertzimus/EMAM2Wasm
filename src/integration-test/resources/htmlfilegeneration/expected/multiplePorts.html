<!doctype html>
<html lang="en-us">
<style>
  .model {
    overflow: hidden;
    width: fit-content;
  }

  .ports {
    overflow: hidden;
  }

  .inports {
    float: left;
    margin: 10px;
  }

  .outports {
    float: left;
    margin: 10px;
  }

  .inport-header {
    margin: 0 auto;
    width: fit-content;
  }

  .inport-label {
    float: left;
    margin-right: 5px;
  }

  .inport-field {
    float: right;
    width: 200px;
  }

  .outport-header {
    margin: 0 auto;
    width: fit-content;
  }

  .outport-label {
    float: left;
    margin-right: 5px;
  }

  .outport-field {
    float: left;
    min-width: 200px;
  }

  .execute {
    width: fit-content;
    margin: 10px auto;
  }

  .error {
    color: red;
    margin: 10px;
  }

  .inport-container-m1 {
    overflow: hidden;
    margin: 3px;
  }

  .inport-container-m2 {
    overflow: hidden;
    margin: 3px;
  }

  .inport-container-c {
    overflow: hidden;
    margin: 3px;
  }

  .outport-container-v {
    overflow: hidden;
    margin: 3px;
  }

  .outport-container-k {
    overflow: hidden;
    margin: 3px;
  }
</style>
<head>
  <script type="text/javascript"
          src="https://cdnjs.cloudflare.com/ajax/libs/mathjs/4.0.1/math.min.js"></script>
  <script type="text/javascript" src="multiplePorts_wrapper.js"></script>
  <script type="text/javascript" src="multiplePorts.js"></script>
  <title>Model</title>
</head>
<body>
<div class="container">
  <div class="model">
    <div class="ports">
      <div class="inports">
        <div class="inport-header">
          <h3>Inports</h3>
        </div>
        <div class="inport-container-m1">
          <div class="inport-label">
            m1:
          </div>
          <div class="inport-field">
            <input type="text" id="inport-field-m1">
          </div>
        </div>
        <div class="inport-container-m2">
          <div class="inport-label">
            m2:
          </div>
          <div class="inport-field">
            <input type="text" id="inport-field-m2">
          </div>
        </div>
        <div class="inport-container-c">
          <div class="inport-label">
            c:
          </div>
          <div class="inport-field">
            <input type="text" id="inport-field-c">
          </div>
        </div>
      </div>
      <div class="outports">
        <div class="outport-header">
          <h3>Outports</h3>
        </div>
        <div class="outport-container-v">
          <div class="outport-label">
            v:
          </div>
          <div class="outport-field" id="outport-field-v"></div>
        </div>
        <div class="outport-container-k">
          <div class="outport-label">
            k:
          </div>
          <div class="outport-field" id="outport-field-k"></div>
        </div>
      </div>
    </div>
    <div class="execute">
      <button id="execute" onclick="exec()">Execute</button>
    </div>
  </div>
  <div class="error" id="error"></div>
</div>

<script>
    function exec() {
        document.getElementById("outport-field-v").innerText = "";
        document.getElementById("outport-field-k").innerText = "";
        document.getElementById("error").innerHTML = "";

        try {
            setM1(document.getElementById("inport-field-m1").value);
            setM2(document.getElementById("inport-field-m2").value);
            setC(document.getElementById("inport-field-c").value);

            execute();

            document.getElementById("outport-field-v").innerText = getV();
            document.getElementById("outport-field-k").innerText = getK();
        }
        catch (err) {
            if (err.message === undefined) {
                document.getElementById("error").innerText = err;
            }
            else {
                document.getElementById("error").innerHTML = err.message;
            }
        }
    }
</script>
</body>
</html>
